Dockerfile-integration-tests

```diff
@@ -6,8 +6,13 @@ WORKDIR /app
 RUN npm install
 COPY tsconfig.json /app/tsconfig.json
 COPY typings/ /app/typings/
-COPY source/ /app/source/
+COPY source/contracts/ /app/source/contracts/
+COPY source/deployment/CompileContracts.ts /app/source/deployment/CompileContracts.ts
+COPY source/libraries/CompileSolidity.ts /app/source/libraries/CompileSolidity.ts
```

Comment:
Why are you copying only these files and then copying everything in source later on?

Reply:
We only need contracts and these two files to compile the contracts and store the compiled results.  Docker will cache this and only recompile all of the contracts if one of the files copied above the contract compilation line changes.  This allows us to iterate on the integration tests without having to re-compile all of the contracts every run.

Reply:
Ok, I figured it had to do with caching the contracts.  So Docker's cache works on a line-by-line basis?

Reply:
Yeah, each Docker line is another cache layer.  If you do something like this though, its considered one line:
```
RUN echo hello \
    && echo world
```

Reply:
Yeah, mean I've seen it cache certain parts of the file that it's already run.  It just hadn't occurred to me that we could do the caching in this way so that if non-contract files are modified, it won't recompile.  Good idea!